EIPS/eip-2253.md

```diff
@@ -0,0 +1,172 @@
+---
+eip: 2253
+title: wallet_getAddressBook JSON-RPC Method
+author: Loredana Cirstea (@loredanacirstea)
+discussions-to: https://ethereum-magicians.org/t/eip-2253-add-wallet-getaddressbook-json-rpc-method/3592
+status: Draft
+type: Standards Track
+category: ERC
+created: 2019-08-27
+requires: 1474
+---
+
+## Simple Summary
+
+This is a proposal for a new JSON-RPC call for retrieving a selection of items from a wallet's address book, with the user's permission.
+
+## Abstract
+
+There is no standardized way for a dApp to request address book information from a user's wallet. This leads to either nonoptimal UX, where the user needs to find and copy-paste an Ethereum address, or to duplication of effort across dApps and decreased privacy.
+
+## Motivation
+
+Address books for wallets are becoming a standard feature, greatly improving usability, while maintaining user ownership on this data.
+
+There is no standardized way for a dApp to leverage this information, with the user's permission. dApps can use the `wallet_getAddressBook` JSON-RPC method to retrieve strictly the information that they need from a wallet, in order to process the request.
+
+The current proposal will avoid duplication of effort - dApps will not need to ask the user to create another address book with them. And it also protects the user's privacy, while increasing usability.
+
+## Specification
+
+New JSON-RPC method: `wallet_getAddressBook`.
+
+**Arguments:**
+- filter object, optional:
+  - `count` - type `uint256`, the number of address book records requested; optional - when missing, the entire address book is requested
+  - `fields` - type `string[]`, address book record fields that are requested; optional - when missing, all fields are requested
+
+**Result:**
+- array with address book records
+  - `address` - type `address`, required
+  - `name` - type `string`, optional, may not be returned if user refuses
+  - `icon` - type `base64`, token icon; optional
+
+### Examples
+
+**1) A request to return the entire address book, with all fields:**
+```
+{
+  "id":1,
+  "jsonrpc": "2.0",
+  "method": "wallet_getAddressBook",
+  "params": []
+}
+```
+Result:
+
+```
+{
+  "id":1,
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "address": "0x0000000000000000000000000000000000000001",
+      "name": "My Friend",
+      "icon": "data:image/gif;base64,R0lGODlhAQABAIABAP///wAAACH5BAEKAAEALAAAAAABAAEAAAICTAEAOw=="
+    },
+    {
+      "address": "0x0000000000000000000000000000000000000002",
+      "name": "Beer Buddy"
+    }
+  ]
+}
+```
+
+**2) A request to return one address book item, with all fields:**
+```
+{
+  "id":1,
+  "jsonrpc": "2.0",
+  "method": "wallet_getAddressBook",
+  "params": [
+    {
+      "count": 1,
```

Comment:
This is an ambiguous IMO. What does "one entry" mean, the first one (according to what ordering?), or just some arbitrary entry?

Reply:
Explained in https://github.com/ethereum/EIPs/pull/2253/files#r320375982 - I will amend the spec after I get your opinion on this.